### PR TITLE
Added docs for mcmt gate

### PR DESCRIPTION
**Pull request by tigerjack**
### Summary
Added documentation for the `Aer` `mcmt` gate as per https://github.com/Qiskit/qiskit-aqua/pull/341

### CODE REVIEW

**Comment by tigerjack** [comment]
@hushaohan here.

File: docs/aqua/algorithms.rst [hunk 1]
```diff
@@ -119,6 +119,36 @@ quantum algorithms:
     additionally two angles ``phi`` and ``lam`` for the mcu3.
 
 
+.. _mcmt:
+
+.. topic:: Multiple-Control Multiple-Target (MCMT) Operation
+
+    The *Multiple-Control Multiple-Target (mcmt)* operation, as the name suggests,
+    allows to generalize a single-control, single-target gate (such as `cz` to
```
Comment (hushaohan) [inline docs/aqua/algorithms.rst]: the opening `(` isn't closed?

File: docs/aqua/algorithms.rst [hunk 2]
```diff
@@ -119,6 +119,36 @@ quantum algorithms:
...
+    support multiple control qubits and multiple target qubits.
+    In other words, the single-control gate passed as argument is applied to all
+    the target qubits if all the control qubits are active.
+
+    The kind of gate to apply can be passed as a parameter and should be a single
+    control gate already defined for a ``QuantumCircuit`` object (such as
+    ``QuantumCircuit.cz`` or ``QuantumCircuit.ch``).
+
+    Currently, just one implementation strategy is implemented: *basic*. It
+    employs the same strategy adopted for the basic mode of `mct`, i.e.
```
Comment (hushaohan) [inline docs/aqua/algorithms.rst]: almost "the same": the basic `mct` uses `n-2` ancilla as it uses a `ccx` to write to the target rather than a `cx`

**Review by hushaohan — CHANGES_REQUESTED** [review]
a couple of minor points

File: docs/aqua/algorithms.rst (hunk 2 above)
Comment (tigerjack) [reply docs/aqua/algorithms.rst]: @hushaohan ok I'll modify it, adding some other comments. It was late here yesterday :D

**Review by hushaohan — APPROVED** [review]

**Comment by hushaohan** [comment]
@tigerjack thx for addressing my comments!

**Pull request merged**